From c1566fe0680a7555014b81a79fac71c38c2a5619 Mon Sep 17 00:00:00 2001
From: Harb Abdulhamid <harba@codeaurora.org>
Date: Tue, 1 Mar 2016 13:31:45 -0600
Subject: [PATCH 366/447] PNP / ACPI: add ACPI_RESOURCE_TYPE_SERIAL_BUS as a
 valid type

Git-commit: 86e75410f067e4ff09d6a94d6bb6800ec956ccfe
Patch-mainline: v4.6-rc1
References: fate#320512

An error message is printed for resources of type 19, which is a valid
supported resource type.  The Firmware Test Suite tool (fwts) reports
this as a test failure.  This change fixes the false test failures
for ASL that use type 19 (ACPI_RESOURCE_TYPE_SERIAL_BUS) resources.

Signed-off-by: Harb Abdulhamid <harba@codeaurora.org>
Signed-off-by: Timur Tabi <timur@codeaurora.org>
Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 drivers/pnp/pnpacpi/rsparser.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/pnp/pnpacpi/rsparser.c b/drivers/pnp/pnpacpi/rsparser.c
index 0579649..4b717c6 100644
--- a/drivers/pnp/pnpacpi/rsparser.c
+++ b/drivers/pnp/pnpacpi/rsparser.c
@@ -252,6 +252,10 @@ static acpi_status pnpacpi_allocated_resource(struct acpi_resource *res,
 	case ACPI_RESOURCE_TYPE_GENERIC_REGISTER:
 		break;
 
+	case ACPI_RESOURCE_TYPE_SERIAL_BUS:
+		/* serial bus connections (I2C/SPI/UART) are not pnp */
+		break;
+
 	default:
 		dev_warn(&dev->dev, "unknown resource type %d in _CRS\n",
 			 res->type);
-- 
2.11.0

